sentence.go

```diff
@@ -155,37 +238,39 @@ func MustRegisterParser(sentenceType string, parser ParserFunc) {
 
 // RegisterParser register a custom parser
 func RegisterParser(sentenceType string, parser ParserFunc) error {
-	customParsersMu.Lock()
-	defer customParsersMu.Unlock()
+	defaultSentenceParserMu.Lock()
+	defer defaultSentenceParserMu.Unlock()
 
-	if _, ok := customParsers[sentenceType]; ok {
+	if _, ok := defaultSentenceParser.CustomParsers[sentenceType]; ok {
 		return fmt.Errorf("nmea: parser for sentence type '%q' already exists", sentenceType)
 	}
 
-	customParsers[sentenceType] = parser
+	defaultSentenceParser.CustomParsers[sentenceType] = parser
 	return nil
 }
 
 // Parse parses the given string into the correct sentence type.
 func Parse(raw string) (Sentence, error) {
-	s, err := parseSentence(raw)
+	defaultSentenceParserMu.Lock()
+	defer defaultSentenceParserMu.Unlock()
+
+	return defaultSentenceParser.Parse(raw)
+}
+
+// Parse parses the given string into the correct sentence type.
+// This method is not co-routine safe!
+func (p *SentenceParser) Parse(raw string) (Sentence, error) {
+	s, err := p.ParseBaseSentence(raw)
 	if err != nil {
 		return nil, err
 	}
 
 	// Custom parser allow overriding of existing parsers
-	if parser, ok := customParsers[s.Type]; ok {
+	if parser, ok := p.CustomParsers[s.Type]; ok {
```

Comment:
`CustomParsers` could be `nil`.

Reply:
this is ok for maps.  See https://go.dev/play/p/LyUUR3faptx